pyregion/wcs_converter.py

```diff
@@ -1,76 +1,135 @@
 import copy
 
-from .wcs_helper import estimate_cdelt, estimate_angle
-from .region_numbers import CoordOdd, CoordEven, Distance, Angle
+from astropy.coordinates import SkyCoord
+from astropy.wcs import WCS
+from astropy.wcs.utils import proj_plane_pixel_area, proj_plane_pixel_scales
+import numpy as np
+from .wcs_helper import _estimate_angle
+from .region_numbers import CoordOdd, Distance, Angle
 from .parser_helper import Shape, CoordCommand
 from .region_numbers import SimpleNumber, SimpleInteger
 
 
-def convert_to_imagecoord(cl, fl, wcs_proj, sky_to_sky, xy0, rot_wrt_axis=1):
-    fl0 = fl
-    new_cl = []
-
-    while cl:
-        if len(fl) == 0:
-            fl = fl0
-
-        if fl[0] == CoordOdd and fl[1] == CoordEven:
-
-            ll = sky_to_sky([cl[0]], [cl[1]])
-            x0, y0 = wcs_proj.topixel(ll)
-            xy0 = [x0[0], y0[0]]
-            new_cl.extend(xy0)
-            cl = cl[2:]
-            fl = fl[2:]
-        elif fl[0] == Distance:
-            degree_per_pixel = estimate_cdelt(wcs_proj, *xy0)
-            new_cl.append(cl[0] / degree_per_pixel)
-            cl = cl[1:]
-            fl = fl[1:]
-        elif fl[0] == Angle:
-            rot1, rot2 = estimate_angle(wcs_proj, xy0[0], xy0[1], sky_to_sky)
-            if rot_wrt_axis == 1:
-                # use the angle between the X axis and North
-                new_cl.append(cl[0] + rot1 - 180.)
-            elif rot_wrt_axis == 2:
-                # use the angle between the Y axis and North
-                new_cl.append(cl[0] + rot2 - 90.)
+def _generate_arg_types(coordlist_length, shape_name):
+    """Find coordinate types based on shape name and coordlist length
+
+    This function returns a list of coordinate types based on which
+    coordinates can be repeated for a given type of shap
+
+    Parameters
+    ----------
+    coordlist_length : int
+        The number of coordinates or arguments used to define the shape.
+
+    shape_name : str
+        One of the names in `pyregion.ds9_shape_defs`.
+
+    Returns
+    -------
+    arg_types : list
+        A list of objects from `pyregion.region_numbers` with a length equal to
+        coordlist_length.
+
+    """
+    from .ds9_region_parser import ds9_shape_defs
+
+    initial_arg_types = ds9_shape_defs[shape_name].args_list
+    arg_repeats = ds9_shape_defs[shape_name].args_repeat
+
+    if arg_repeats is None:
+        return initial_arg_types
+    else:
```

Comment:
Remove the else and de-dent the code block.
Result will be the same, but IMO a bit nicer to read.